Python/sysmodule.c

```diff
@@ -1079,9 +1079,6 @@ trace_trampoline(PyObject *self, PyFrameObject *frame,
     if (result != Py_None) {
         Py_XSETREF(frame->f_trace, result);
     }
-    else {
-        Py_DECREF(result);
-    }
```

Comment:
Let's hold off on this part.  I'd rather we were more careful about it in a way that also helps in the other places we could drop incref/decref for `None` or other known immortal objects.  I've opened gh-117425 to discuss this.

If we added a macro for this, we'd use it here:

```suggestion
    else {
        Py_DECREF_IMMORTAL(result);
    }
```

Reply:
Okay sure that makes sense.